lua/nvim-surround/config.lua

```diff
@@ -24,54 +21,171 @@ M.default_opts = {
         change = "cs",
     },
     delimiters = {
-        invalid_key_behavior = function()
-            vim.api.nvim_err_writeln(
-                "Error: Invalid character! Configure this message in " .. 'require("nvim-surround").setup()'
-            )
-        end,
-        pairs = {
-            ["("] = { "( ", " )" },
-            [")"] = { "(", ")" },
-            ["{"] = { "{ ", " }" },
-            ["}"] = { "{", "}" },
-            ["<"] = { "< ", " >" },
-            [">"] = { "<", ">" },
-            ["["] = { "[ ", " ]" },
-            ["]"] = { "[", "]" },
-            ["i"] = function()
+        ["("] = {
```

Comment:
Will it be possible to simply set `(` to a table of characters here? While that's basically just syntactic sugar, I think that makes the API look a bit nicer for simple cases like this.

Reply:
The default configuration has to "look complicated", since it uses the internal representation that every function in the program assumes exists. That being said, user configuration will be simpler, as I will write a "translation" function that will convert user-provided configuration into the internal form, and then merge to provide the proper feature set. I think I plan on the `add` key being a table of characters, as it would provide a better "mix-and-match" for syntactic sugar between `add`/`delete`/`change` functions instead of tying everything together into an "all or nothing".

Reply:
Initial work has been done; if the surrounding pair is "static" then you may provide a pair instead of a function. Furthermore, strings will be auto-converted to tables with one string (i.e. multi-line form). For example, I have this in my `ftplugin/lua.lua` for printing (via `yssp`):
```lua
["p"] = {
    add = { "vim.pretty_print(", ")" },
},
```